src/lib/analyses.ts

```diff
@@ -32,54 +150,89 @@ export enum AggregateRecommendationDecision {
   chosenVariationId?: number
+  statisticallySignificant?: boolean
+  practicallySignificant?: PracticalSignificanceStatus
+}
+
+const PracticalSignificanceStatusToDecision: Record<PracticalSignificanceStatus, AggregateRecommendationDecision> = {
+  [PracticalSignificanceStatus.No]: AggregateRecommendationDecision.DeployAnyVariation,
+  [PracticalSignificanceStatus.Uncertain]: AggregateRecommendationDecision.Inconclusive,
+  [PracticalSignificanceStatus.Yes]: AggregateRecommendationDecision.DeployChosenVariation,
 }
 
 /**
- * Returns the aggregate recommendation over analyses of different analysis strategies.
+ * Returns the recommendation for a single analysis.
  *
- * @param analyses Analyses of different strategies for the same day.
- * @param defaultStrategy Default strategy in the context of an aggregateRecommendation..
+ * See file-level recommendation documentation.
  */
-export function getAggregateRecommendation(
-  analyses: Analysis[],
-  defaultStrategy: AnalysisStrategy,
+export function getMetricAssignmentRecommendation(
+  experiment: ExperimentFull,
+  metric: MetricBare,
+  analysis: Analysis,
 ): AggregateRecommendation {
-  const recommendationChosenVariationIds = analyses
-    .map((analysis) => analysis.recommendation)
-    .map((recommendation) => recommendation?.chosenVariationId)
-    .filter((x) => x)
-  const recommendationConflict = [...new Set(recommendationChosenVariationIds)].length > 1
-  if (recommendationConflict) {
+  const metricAssignment = experiment.metricAssignments.find(
+    (metricAssignment) => metricAssignment.metricAssignmentId === analysis.metricAssignmentId,
+  )
+  const diffCredibleIntervalStats =
+    analysis && metricAssignment && getDiffCredibleIntervalStats(analysis, metricAssignment)
+  const analysisStrategy = analysis.analysisStrategy
+  if (!analysis.metricEstimates || !metricAssignment || !diffCredibleIntervalStats) {
     return {
-      decision: AggregateRecommendationDecision.ManualAnalysisRequired,
+      analysisStrategy,
+      decision: AggregateRecommendationDecision.MissingAnalysis,
     }
   }
 
-  const recommendation = analyses.find((analysis) => analysis.analysisStrategy === defaultStrategy)?.recommendation
-  if (!recommendation) {
-    return {
-      decision: AggregateRecommendationDecision.MissingAnalysis,
-    }
+  const { practicallySignificant, statisticallySignificant, isPositive } = diffCredibleIntervalStats
+  const decision = PracticalSignificanceStatusToDecision[practicallySignificant]
+  const defaultVariation = experiment.variations.find((variation) => variation.isDefault) as Variation
+  const nonDefaultVariation = experiment.variations.find((variation) => !variation.isDefault) as Variation
+  let chosenVariationId = undefined
+  if (decision === AggregateRecommendationDecision.DeployChosenVariation) {
+    chosenVariationId =
+      isPositive === metric.higherIsBetter ? nonDefaultVariation.variationId : defaultVariation.variationId
```

Comment:
Just a couple of other ways. Not fussed about this. :slightly_smiling_face: 

No need to go through the variations if we don't use the result:
```suggestion
  let chosenVariationId = undefined
  if (decision === AggregateRecommendationDecision.DeployChosenVariation) {
    const defaultVariation = experiment.variations.find((variation) => variation.isDefault) as Variation
    const nonDefaultVariation = experiment.variations.find((variation) => !variation.isDefault) as Variation
    chosenVariationId =
      isPositive === metric.higherIsBetter ? nonDefaultVariation.variationId : defaultVariation.variationId
```

And we don't need to iterate twice:
```suggestion
  let chosenVariationId = undefined
  if (decision === AggregateRecommendationDecision.DeployChosenVariation) {
    const chooseDefaultVariation = isPositive !== metric.higherIsBetter
    const chosenVariation = experiment.variations.find((variation) => variation.isDefault === chooseDefaultVariation) as Variation
    chosenVariationId = chosenVariation.variationId
```

Reply:
I would actually like to rename `chosenVariationId` to be something like `preferredVariationId` or `betterVariationId` and include it when statistically significant, this would allow us to have better health monitoring. We might also want to trigger manual analysis required for cases where there are conflicts of statistical significance, this would be a step in-between what we currently have of conflicting practical significance and what we had before - conflicting practical significance including not practical significant.